internal/assignid.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"fmt"
+
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+)
+
+func (conv *Conv) AssignIdToSourceSchema() {
+	conv.AssignTableId()
+	conv.AssignColumnId()
+	conv.AssignPkId()
+	conv.AssignIndexId()
+	conv.AssginFkId()
+}
+
+func (conv *Conv) AssignTableId() {
+	srcSchema := map[string]schema.Table{}
+
+	for _, v := range conv.SrcSchema {
+		tableId := GenerateTableId()
+		srcTable := v
+		srcTable.Id = tableId
+		srcSchema[tableId] = srcTable
+	}
+	conv.SrcSchema = nil
+	conv.SrcSchema = srcSchema
+
+}
+
+func (conv *Conv) AssignColumnId() {
+	for tableId, table := range conv.SrcSchema {
+
+		colNames := make([]string, 0, len(table.ColDefs))
+		for colName := range table.ColDefs {
+			colNames = append(colNames, colName)
+		}
```

Comment:
We can use the ColIds list which has Col names at this point instead of scanning through the maps

Reply:
Updated